### PR TITLE
Review PR

### PR DESCRIPTION
**Pull request by woop**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+feast_custom_provider/
```
Comment (achals) [inline requirements.txt:1]: This is cool, didn't know we could specify subdirs in the requirements.txt file.

File: basic_feature_repo/repo.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    event_timestamp_column="event_timestamp",
+    created_timestamp_column="created",
+)
+
+driver = Entity(name="driver_id", value_type=ValueType.INT64, description="driver id",)
+
+driver_hourly_stats_view = FeatureView(
+    name="driver_hourly_stats",
+    entities=["driver_id"],
+    ttl=timedelta(days=365),
```
Comment (achals) [inline basic_feature_repo/repo.py:16]: btw I'm pretty sure this is a countdown to failing tests a year from now :) Not sure if there's an alternative though. You could make the TTL very large for this case safely I think.

File: feast_custom_provider/custom_provider.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.registry import Registry
+from feast.repo_config import RepoConfig
+from tqdm import tqdm
+
+
+class MyCustomProvider(LocalProvider):
+    def __init__(self, config: RepoConfig, repo_path):
+        super().__init__(config)
+        # Add your custom init code here. This code runs on every
```
Comment (achals) [inline feast_custom_provider/custom_provider.py]: typo, dangling sentence

File: test_custom_provider.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+def test_end_to_end():
+    fs = FeatureStore("basic_feature_repo/")
+
+    # apply repository
+    fs.apply([driver, driver_hourly_stats_view])
+
+    # load data into online store
+    fs.materialize_incremental(end_date=datetime.now())
+
+    # Read features from online store
```
Comment (achals) [inline test_custom_provider.py:18]: Wanna add a call to `fs.get_historical_features()` just for completeness?

**Review by achals — CHANGES_REQUESTED** [review]
minor nits

File: basic_feature_repo/repo.py (hunk 2 above)
Comment (woop) [reply basic_feature_repo/repo.py:16]: That's ok, we can optimize in a year. I dont even think providers will be around then.

**Review by achals — APPROVED** [review]

**Pull request closed without merging**